### PR TITLE
Fix all remaining LGTM warnings in AMP code

### PR DESCRIPTION
**Pull request by rsimha-amp**
Fixes the following LGTM warnings:

- [extensions/amp-lightbox-viewer/0.1/service/lightbox-manager-impl.js#L218](https://lgtm.com/projects/g/ampproject/amphtml/snapshot/279db9d5a15d63c53a4de7a24efc58f9a0729a0b/files/extensions/amp-lightbox-viewer/0.1/service/lightbox-manager-impl.js?sort=name&dir=ASC&mode=heatmap&excluded=false#L218)
- [extensions/amp-web-push/0.1/amp-web-push.service-worker.js#L74](https://lgtm.com/projects/g/ampproject/amphtml/snapshot/279db9d5a15d63c53a4de7a24efc58f9a0729a0b/files/extensions/amp-web-push/0.1/amp-web-push.service-worker.js#L74)
- [extensions/amp-story/0.1/share.js#L106](https://lgtm.com/projects/g/ampproject/amphtml/snapshot/279db9d5a15d63c53a4de7a24efc58f9a0729a0b/files/extensions/amp-story/0.1/share.js?sort=name&dir=ASC&mode=heatmap&excluded=false#L106)
- [extensions/amp-bind/0.1/bind-impl.js#L938](https://lgtm.com/projects/g/ampproject/amphtml/snapshot/279db9d5a15d63c53a4de7a24efc58f9a0729a0b/files/extensions/amp-bind/0.1/bind-impl.js?sort=name&dir=ASC&mode=heatmap&excluded=false#L938)
- [extensions/amp-image-lightbox/0.1/amp-image-lightbox.js#L908](https://lgtm.com/projects/g/ampproject/amphtml/snapshot/279db9d5a15d63c53a4de7a24efc58f9a0729a0b/files/extensions/amp-image-lightbox/0.1/amp-image-lightbox.js?sort=name&dir=ASC&mode=heatmap&excluded=false#L908)

With this, there should be zero remaining [LGTM warnings and errors](https://lgtm.com/projects/g/ampproject/amphtml/alerts/?mode=list) in the AMP code (until new ones are checked in :/)

Partial fix for #12597

### CODE REVIEW

**Comment by rsimha-amp** [comment]
@cramforce With this PR, all existing errors and warnings should be eliminated from our code: https://lgtm.com/projects/g/ampproject/amphtml/alerts/?mode=list

Perhaps now is the time to allow the lgtm tool to add a comment to PRs when it finds new errors / warnings? See https://github.com/ampproject/amphtml/issues/10092#issuecomment-356788056

File: extensions/amp-story/0.1/share.js [hunk 1]
```diff
@@ -103,7 +103,7 @@ function buildProviderParams(opt_params) {
   const attrs = dict();
 
   if (opt_params) {
-    Object.keys(opt_params || {}).forEach(field => {
+    Object.keys(opt_params).forEach(field => {
```
Comment (jridgewell) [inline extensions/amp-story/0.1/share.js]: This is an error, `undefined` (due to the `=` in the type def) cannot be passed to `Object.keys`.

File: extensions/amp-web-push/0.1/amp-web-push.service-worker.js [hunk 2]
```diff
@@ -67,17 +67,17 @@ self.addEventListener('message', event => {
     - payload: An optional JavaScript object containing extra data relevant to
       the command.
    */
-  const {command, payload} = event.data;
+  const {command, payload} = event.data; // eslint-disable-line no-unused-vars
```
Comment (jridgewell) [inline extensions/amp-web-push/0.1/amp-web-push.service-worker.js]: `payload` can just be dropped.

Comment (rsimha-amp) [reply extensions/amp-web-push/0.1/amp-web-push.service-worker.js]: Fixed.

File: extensions/amp-story/0.1/share.js (hunk 1 above)
Comment (rsimha-amp) [reply extensions/amp-story/0.1/share.js]: The `if (opt_params)` in the above line already prevents `undefined` from being passed to `Object.keys`.

Comment (rsimha-amp) [reply extensions/amp-story/0.1/share.js]: @jridgewell PTAL

Comment (jridgewell) [reply extensions/amp-story/0.1/share.js]: Ah, didn't see that.

**Comment by sjvs** [comment]
Hi @rsimha-amp! Great work getting all errors reported by lgtm.com out of amphtml! Enabling comments is indeed a great way to make sure no new alerts sneak back in. Any repo admin can enable the comments here: https://lgtm.com/projects/g/ampproject/amphtml/ci/. I believe @cramforce was the one who set the integration up initially?

Full disclosure: I'm part of the team behind lgtm.com. Let me know if we can do anything to help!

**Pull request merged**